components/action-mgt/org.wso2.carbon.identity.action.management/src/main/java/org/wso2/carbon/identity/action/management/dao/impl/ActionManagementDAOImpl.java

```diff
@@ -224,6 +256,81 @@ public Action getActionByActionId(String actionId, Integer tenantId) throws Acti
         }
     }
 
+    @Override
+    public Action updateActionEndpointAuthProperties(String actionId, AuthType authentication, int tenantId)
+            throws ActionMgtException {
+
+        Connection dbConnection = IdentityDatabaseUtil.getDBConnection(true);
+        try {
+            Map<String, String> nonSecretEndpointProperties = authentication.getProperties().stream()
+                    .filter(property -> !property.getIsConfidential())
+                    .collect(Collectors.toMap(AuthProperty::getName, AuthProperty::getValue));
+            // Update non-secret endpoint properties.
+            updateActionEndpointProperties(dbConnection, actionId, nonSecretEndpointProperties, tenantId);
+            // Encrypt and update non-secret endpoint properties.
+            actionSecretProcessor.encryptAssociatedSecrets(authentication.getProperties(), actionId,
+                    authentication.getType().name());
+            IdentityDatabaseUtil.commitTransaction(dbConnection);
+
+            return getActionByActionId(actionId, tenantId);
+        } catch (ActionMgtException | SecretManagementException e) {
+            if (LOG.isDebugEnabled()) {
+                LOG.debug("Error while updating the action secrets of Auth type: " + authentication.getType() +
+                        " of Action id: " + actionId + " in tenantDomain: " +
+                        IdentityTenantUtil.getTenantDomain(tenantId) +
+                        ". Rolling back updated action endpoint authentication properties.");
+            }
+            IdentityDatabaseUtil.rollbackTransaction(dbConnection);
+            throw ActionManagementUtil.handleServerException(
+                    ActionMgtConstants.ErrorMessages.ERROR_WHILE_UPDATING_ENDPOINT_PROPERTIES, e);
+        } finally {
+            IdentityDatabaseUtil.closeConnection(dbConnection);
+        }
+    }
+
+    @Override
+    public Action updateActionEndpoint(String actionType, String actionId, EndpointConfig endpoint,
+                                       AuthType currentAuthentication, int tenantId)
+            throws ActionMgtException {
+
+        Connection dbConnection = IdentityDatabaseUtil.getDBConnection(true);
+        try (NamedPreparedStatement statement = new NamedPreparedStatement(dbConnection,
+                ActionMgtSQLConstants.Query.DELETE_ACTION_ENDPOINT_PROPERTIES)) {
+
+            statement.setString(ActionMgtSQLConstants.Column.ACTION_ENDPOINT_UUID, actionId);
+            statement.setInt(ActionMgtSQLConstants.Column.TENANT_ID, tenantId);
+            statement.executeUpdate();
```

Comment:
Why do we want to delete the endpoint to update the auth type ?

Reply:
Since endpoint uri is also stored in the IDN_ACTION_ENDPOINT table its easier to delete all the values and add them again.